src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/start-interaction.mdx

```diff
@@ -0,0 +1,623 @@
+- /docs/mobile-monitoring/new-relic-mobile-android/startinteraction-android-sdk-api
+- /docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/startinteraction-android-sdk-api
+- /docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/startinteraction-android-agent-api
+- /docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/startinteraction
+- /docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/startInteraction
+- /docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/start-interaction
+---
+
+<Tabs>
+	<TabsBar>
+        <TabsBarItem id="android">
+            Android
+        </TabsBarItem>
+        <TabsBarItem id="ios">
+            iOS
+        </TabsBarItem>
+        <TabsBarItem id="capacitor">
+            Capacitor
+        </TabsBarItem>
+        <TabsBarItem id="cordova">
+            Cordova
+        </TabsBarItem>
+        <TabsBarItem id="flutter">
+            Flutter
+        </TabsBarItem>
+        <TabsBarItem id="react">
+            React Native
+        </TabsBarItem>
+        <TabsBarItem id="xamarin">
+            Xamarin
+        </TabsBarItem>
+        <TabsBarItem id="maui">
+            .NET MAUI
+        </TabsBarItem>
+    </TabsBar>
+
+    <TabsPages>
+        <TabsPageItem id="android">
+## Syntax [#syntax]
+
+### Java [#java]
+
+```java
+NewRelic.startInteraction(string $interactionName)
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+NewRelic.startInteraction(actionName: String)
+```
+## Description [#description]
+
+Create an interaction to instrument a method in your Android app code.
+
+To name an interaction that already exists and is already being tracked, see `setInteractionName()`
+
+## Parameters [#parameters]
+
+<table>
+  <thead>
+    <tr>
+      <th width={200}>
+        Parameter
+      </th>
+      <th width={200}>
+        Type
+      </th>
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$interactionName`
+      </td>
+
+            <td>
+        `string`
+      </td>
+
+      <td>
+        Required. The name you want to give to the interaction.
+      </td>
+    </tr>
+
+  </tbody>
+</table>
+
+## Return values [#return-values]
+
+Returns an interaction ID numberw hich can be used for [ending the interaction](/docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/end-interaction) at a certain point.
+
+## Example [#example]
+
+Here's an xample of starting to track an interaction named `RefreshContacts`:
+
+### Java [#java]
+
+```java
+public class MainActivity extends Activity {
+  ...
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    switch (item.getItemId()) {
+      case R.id.action_refresh:
+        NewRelic.startInteraction("RefreshContacts");
+   ……
+        return true;
+      default:
+        return super.onOptionsItemSelected(item);
+    }
+  }
+  ...
+}
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+
+```
+
```

Comment:
@ndesai-newrelic Can you please provide a kotlin example equivalent to the Java example?

Reply:
```suggestion
```kotlin

```
class MainActivity : AppCompatActivity() {

  .....
    var client: OkHttpClient = OkHttpClient();


        binding.fab.setOnClickListener { view ->

            val interActionId = NewRelic.startInteraction("Getting Data From Server")

                lifecycleScope.launch(Dispatchers.IO) {
                    val result = getRequest()
                    NewRelic.endInteraction(interActionId)
                }
            }
        }
    ....
    }
```